src/containers/Header/Search.tsx

```diff
@@ -32,66 +42,98 @@ const determineHashType = async (id: string, rippledContext: XrplClient) => {
 // separator for currency formats
 const separators = /[.:+-]/
 
-const getIdType = async (id: string, rippledContext: XrplClient) => {
+const getRoute = async (
+  id: string,
+  rippledContext: XrplClient,
+): Promise<{ type: string; path: string } | null> => {
   if (DECIMAL_REGEX.test(id)) {
-    return 'ledgers'
+    return {
+      type: 'ledgers',
+      path: buildPath(LEDGER_ROUTE, { identifier: id }),
+    }
   }
   if (isValidClassicAddress(id)) {
-    return 'accounts'
+    return {
+      type: 'accounts',
+      path: buildPath(ACCOUNT_ROUTE, { id: normalizeAccount(id) }),
+    }
   }
   if (HASH_REGEX.test(id)) {
     // Transactions and NFTs share the same syntax
     // We must make an api call to ensure if it's one or the other
-    return determineHashType(id, rippledContext)
+    const type = await determineHashType(id, rippledContext)
+    let path
+    if (type === 'transactions') {
+      path = buildPath(TRANSACTION_ROUTE, { identifier: id.toUpperCase() })
+    } else if (type === 'nft') {
+      path = buildPath(NFT_ROUTE, { id: id.toUpperCase() })
+    }
+
+    return {
+      path,
+      type,
+    }
   }
   if (isValidXAddress(id) || isValidClassicAddress(id.split(':')[0])) {
-    return 'accounts' // TODO: Consider a new path/page specific to X-addresses
+    return {
+      type: 'accounts',
+      path: buildPath(ACCOUNT_ROUTE, { id: normalizeAccount(id) }), // TODO: Consider a new path/page specific to X-addresses
+    }
   }
   if (isValidPayString(id) || isValidPayString(id.replace('@', '$'))) {
-    return 'paystrings'
+    let normalizedId = id
+    if (!isValidPayString(id)) {
+      normalizedId = id.replace('@', '$')
+    }
+
+    return {
+      type: 'paystrings',
+      path: buildPath(PAYSTRING_ROUTE, { id: normalizedId }),
+    }
   }
   if (
     (CURRENCY_REGEX.test(id) || FULL_CURRENCY_REGEX.test(id)) &&
     isValidClassicAddress(id.split(separators)[1])
   ) {
-    return 'token'
+    const components = id.split(separators)
+    return {
+      type: 'token',
+      path: buildPath(TOKEN_ROUTE, {
+        token: `${components[0].toLowerCase()}.${components[1]}`,
+      }),
+    }
   }
   if (VALIDATORS_REGEX.test(id)) {
-    return 'validators'
+    return {
+      type: 'validators',
+      path: buildPath(VALIDATOR_ROUTE, { identifier: normalizeAccount(id) }),
```

Comment:
What's the difference between the `id` and `identifier` parameters to these paths? (Maybe it'd be worth making them consistent in a separate PR? Or giving them more specific names if they're distinct in an important way?)

Reply:
Agreed.